test/k8sT/Services.go

```diff
@@ -82,24 +117,29 @@ var _ = Describe("K8sServicesTest", func() {
 		kubectl.Apply(demoDSPath)
 		defer kubectl.Delete(demoDSPath)
 
-		pods, err := kubectl.WaitforPods("default", "-l zgroup=testDS", 300)
-		Expect(pods).Should(BeTrue())
-		Expect(err).Should(BeNil())
+		waitPodsDs()
 
 		svcIP, err := kubectl.Get(
 			"default", "service testds-service").Filter("{.spec.clusterIP}")
 		Expect(err).Should(BeNil())
 		Expect(govalidator.IsIP(svcIP.String())).Should(BeTrue())
-
-		status := kubectl.Node.Exec(fmt.Sprintf("curl http://%s/", svcIP))
-		status.ExpectSuccess()
-
-		k8s2 := helpers.CreateKubectl("k8s2", logger)
-		status = k8s2.Node.Exec(fmt.Sprintf("curl http://%s/", svcIP))
-		status.ExpectSuccess()
+		url := fmt.Sprintf("http://%s/", svcIP)
+		testHTTPRequest(url)
 	})
 
 	//TODO: Check service with IPV6
-	//TODO: NodePort? It is ready?
 
+	It("Check NodePort", func() {
+		demoDSPath := fmt.Sprintf("%s/demo_ds.yaml", kubectl.ManifestsPath())
+		kubectl.Apply(demoDSPath)
+		defer kubectl.Delete(demoDSPath)
+
+		waitPodsDs()
+
+		var data v1.Service
+		err := kubectl.Get("default", "service test-nodeport").UnMarshal(&data)
+		Expect(err).Should(BeNil(), "Can not retrieve service")
+		url := fmt.Sprintf("http://%s:%d", data.Spec.ClusterIP, data.Spec.Ports[0].Port)
```

Comment:
Use https://golang.org/pkg/net/#JoinHostPort

Reply:
should be ok now ;-)